internal/utils/utils.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// ErrNoNamespace indicates that a namespace could not be found for the current
+// environment
+var ErrNoNamespace = fmt.Errorf("namespace not found for current environment")
+
+// GetOperatorNamespace returns the namespace the operator should be running in.
+func GetOperatorNamespace(readNamespace func() ([]byte, error)) (string, error) {
```

Comment:
Alternative approach:
Instead of doing
```go
func GetOperatorNamespace(readNamespace func() ([]byte, error)) (string, error) {
	...
}

var readNamespace = func() ([]byte, error) {
	return ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
}

func foo() {
	_, _ = GetOperatorNamespace(readNamespace)
}
```

you could do

```go
func GetOperatorNamespace() (string, error) {
	...
}

var readNamespace = GetOperatorNamespace()

func foo() {
	_, _ = readNamespace()
}
```

which is a little less confusing and still mockable in tests in a similar way as to how you have it now:

```go
var _ = Describe("test namespace", func() {
	// This
	readNamespace = func() (string, error) {
		return "some-ns"
	}
	// Or this
	f := func(s string) (func() (string, error)) {
		return func() (string, error) { return s, nil }
	}
	readNamespace = f("some-ns")
}
```